extensions/PhabBugz/lib/Project.pm

```diff
@@ -142,12 +146,20 @@ sub BUILDARGS {
 #########################
 
 sub create {
-    my ( $class, $params ) = @_;
-
-    my $name = trim( $params->{name} );
-    $name || ThrowCodeError( 'param_required', { param => 'name' } );
+    state $check = compile(
+        $Invocant | ClassName,
+        Dict[
+            name => Str,
+            description => Str,
+            view_policy => Str,
+            edit_policy => Str,
+            join_policy => Str,
+        ]
+    );
+    my ( $class, $params ) = $check->(@_);
 
-    my $description = $params->{description} || 'Need description';
+    my $name        = $params->{name};
```

Comment:
The old code called trim on the name, just incase you overlooked that.

Reply:
That is a good catch, I'll update it to still call trim. The other change here isn't needed because the type checking prevents those cases from happening.